c7n/resources/ec2.py

```diff
@@ -1022,7 +1022,7 @@ def process(self, resources, event=None):
 
 
 @actions.register('set-monitoring')
-class MonitorInstances(BaseAction, StateTransitionFilter):
```

Comment:
The class definitions could be removed alongside these changes too, like on line 243 of this file.

Reply:
That couldn't be removed because it's still used for the filters in this program.

Reply:
I've copied `filter_resources` from `core.actions` to `core.filters`.  It doesn't feel quite right since it's duplicated code, but they don't inherit from another common class, so it was unavoidable for now.  Should I eliminate this redundancy or just leave it to be handled as part of another refactor?

Some projects like to apply PRs that are more atomic and easier to review rather than getting all the changes in one PR.

Thanks!

Reply:
Thanks for checking on them. I agree, it would be nice to get it into one location. Maybe there could be an implementation that both inherit from. @kapilt, thoughts? Would something like this fit into utils?

Reply:
Resolved via 9c21cfe